instant-xml-macros/src/lib.rs

```diff
@@ -6,15 +6,51 @@ use syn::parse_macro_input;
 
 #[proc_macro_derive(ToXml)]
 pub fn to_xml(input: TokenStream) -> TokenStream {
-    let ast = parse_macro_input!(input as syn::ItemStruct);
+    let ast = parse_macro_input!(input as syn::DeriveInput);
     let ident = &ast.ident;
-    let name = ident.to_string();
+    let root_name = ident.to_string();
+    let mut output: proc_macro2::TokenStream = TokenStream::from(quote!(
+        ::std::format!("<{}>", #root_name)
+    )).into();
+
+    match &ast.data {
+        syn::Data::Enum(_) => {
+            todo!();
+        }
+        syn::Data::Struct(ref data) => {
+            match data.fields {
+                syn::Fields::Named(ref fields) => {
+                    fields
+                    .named
+                    .iter()
+                    .for_each(|field| {
+                        let field_name = field.ident.as_ref().unwrap().to_string();
+                        let field_value = field.ident.as_ref().unwrap();
+                        output = TokenStream::from(quote!(
+                            ::std::format!("{}<{}>{}</{}>", #output, #field_name, self.#field_value, #field_name)
+                        )).into();
+                    });
+                }
+                syn::Fields::Unnamed(_) => {
```

Comment:
You can write a simple branch like this as `todo!(),` instead of `{ todo!(); }`.

Reply:
done